### ISSUE TITLE
[CI] 7.17 unix compatibility test for amazon-2022 failing

### ISSUE THREAD

**Issue by slobodanadamovic** [elastic/elasticsearch#89646]
### CI Link

https://gradle-enterprise.elastic.co/s/dg6kjh3dscczw

### Repro line

run `:qa:verify-version-constants:v6.8.24#integTest` on Amazon Linux 2022

### Does it reproduce?

Didn't try

### Applicable branches

7.17

### Failure history

https://gradle-enterprise.elastic.co/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Execution%20failed%20for%20task%20*%0A%3E%20process%20was%20found%20dead%20while%20waiting%20for%20ports%20files%2C%20*&search.timeZoneId=Europe/Zurich

### Failure excerpt

The main cause:

```
» [2022-08-26T09:16:35,768][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [v6.8.24-0] uncaught exception in thread [main]
»  org.elasticsearch.bootstrap.StartupException: ElasticsearchException[Failure running machine learning native code. This could be due to running on an unsupported OS or distribution, missing OS libraries, or a problem with the temp directory. To bypass this problem by running Elasticsearch without machine learning functionality set [xpack.ml.enabled: false].]
»  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:163) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124) ~[elasticsearch-cli-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:116) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:93) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  Caused by: org.elasticsearch.ElasticsearchException: Failure running machine learning native code. This could be due to running on an unsupported OS or distribution, missing OS libraries, or a problem with the temp directory. To bypass this problem by running Elasticsearch without machine learning functionality set [xpack.ml.enabled: false].
»  	at org.elasticsearch.xpack.ml.MachineLearning.createComponents(MachineLearning.java:440) ~[?:?]
»  	at org.elasticsearch.node.Node.lambda$new$11(Node.java:472) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:269) ~[?:1.8.0_302]
»  	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384) ~[?:1.8.0_302]
»  	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_302]
»  	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_302]
»  	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_302]
»  	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_302]
»  	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[?:1.8.0_302]
»  	at org.elasticsearch.node.Node.<init>(Node.java:475) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.node.Node.<init>(Node.java:266) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:212) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:212) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:333) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-6.8.24-SNAPSHOT.jar:6.8.24-SNAPSHOT]
»  	... 6 more
```

**Comment by droberts195** [comment]
This has stopped failing.  I think it's because the `libxcrypt-compat` package was added to the Amazon Linux 2022 CI image 5 days ago and that's fixed it.

There was an [issue](https://github.com/elastic/ml-cpp/issues/478) that the ML code was linking to a library that wasn't installed by default on modern Linux distributions.  This was fixed in 7.2 by https://github.com/elastic/ml-cpp/pull/480.  The reason this test was failing is that it's trying to do a BWC test with 6.8.24, so needs to run ML version 6.8.24, and that requires the legacy package be installed.  Now the legacy package is installed the BWC test should work.

This is all completely irrelevant to customers who only run Elasticsearch on distributions that are supported for the version they're running, as we will never support versions earlier than 7.2 on Amazon Linux 2022.

**Comment by mark-vieira** [comment]
Thanks for tracking this down @droberts195.
